### PR TITLE
Show group members

### PR DESCRIPTION
**Pull request by tchayen**

### CODE REVIEW

File: client/src/components/ShowGroup.js [hunk 1]
```diff
@@ -16,15 +16,18 @@ const ShowGroup = ({ match, requestGroupMembers, groups }) => {
     requestGroupMembers(match.params.id)
   }, [])
 
+  console.log('groups', groups)
```
Comment (osdnk) [inline client/src/components/ShowGroup.js]: pls

File: client/src/components/ShowGroup.js [hunk 2]
```diff
@@ -16,15 +16,18 @@ const ShowGroup = ({ match, requestGroupMembers, groups }) => {
...
   const group = groups[match.params.id]
 
   if (!group || group.loading) return <div>loading...</div>
   if (group.error) return <div>error</div>
 
+  console.log('g', group)
```
Comment (osdnk) [inline client/src/components/ShowGroup.js]: pls

File: client/src/store/reducers.js [hunk 3]
```diff
@@ -66,7 +66,7 @@ const session = (
 
 const users = (
   state = {
-    list: [],
+    all: {},
```
Comment (osdnk) [inline client/src/store/reducers.js]: no

File: client/src/store/reducers.js [hunk 4]
```diff
@@ -87,7 +87,23 @@ const users = (
         ...state,
       }
     case RECEIVE_ALL_USERS:
-      return { ...state, list: action.users }
+      const all = {}
+      action.users.forEach(user => (all[user._id] = user))
```
Comment (osdnk) [inline client/src/store/reducers.js]: just why

File: client/src/store/reducers.js [hunk 5]
```diff
@@ -87,7 +87,23 @@ const users = (
...
+      return state
+  }
+}
+
+const members = (state = [], action) => {
+  switch (action.type) {
+    case RECEIVE_GROUP_MEMBERS:
+      if (action.members.length === 0) return state
+      return state
+        .filter(existing => existing.groupId !== action.members[0])
```
Comment (osdnk) [inline client/src/store/reducers.js]: pls

File: client/src/store/selectors.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export const getGroupUsers = (state, groupId) =>
+  state.members
+    .filter(member => member.groupId === groupId)
```
Comment (osdnk) [inline client/src/store/selectors.js]: plss

File: client/src/store/reducers.js (hunk 3 above)
Comment (tchayen) [reply client/src/store/reducers.js]: Why

File: client/src/store/reducers.js (hunk 4 above)
Comment (tchayen) [reply client/src/store/reducers.js]: For being awesome:
```js
.map(member => state.users.all[member.userId])
```

**Comment by osdnk** [comment]
I think it's done in a bad way from very scratch and don't wanna have it merged in this state. I advise another approach:
1. There's no state we want to have each user in our web client
2. Thus I suggest using "members" as a kind of abstraction for users, which are it the group coz these users are only ones we're interested in.

**Pull request merged**